### ISSUE TITLE
wgengine/magic: broken on older Linux kernels?

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#6807]
Tailscale is broken on older Synology NAS due to recvmmsg:

```
admin@ds212j:~$ tailscale  version
1.35.85
  tailscale commit: 98daf99775ea7e61ab103bb42e777ac5e80ea422
  go version: go1.19.4-tsdc0ce6324d

admin@ds212j:~$ uname -a
Linux ds212j 2.6.32.12 #25556 Thu Jul 1 14:25:48 CST 2021 armv5tel GNU/Linux synology_88f6281_212j

admin@ds212j:~$ tailscale  status
...
# Health check:
#     - ReceiveIPv4 is not running
#     - ReceiveIPv6 is not running
#     - This is an unstable (development) version of Tailscale; frequent updates and bugs are likely
```
...
```
2022-12-20 17:47:02.74448869 +0000 UTC: Routine: receive incoming v4 - started
2022-12-20 17:47:02.82383548 +0000 UTC: Routine: receive incoming v6 - started
2022-12-20 17:47:03.095672665 +0000 UTC: Failed to receive v4 packet: read udp 0.0.0.0:41641: recvmmsg: function not implemented
2022-12-20 17:47:03.096176723 +0000 UTC: Routine: receive incoming v4 - stopped
2022-12-20 17:47:03.097173345 +0000 UTC: Failed to receive v6 packet: read udp [::]:41641: recvmmsg: function not implemented
2022-12-20 17:47:03.097574948 +0000 UTC: Routine: receive incoming v6 - stopped
2022-12-20 17:47:03.174790769 +0000 UTC: Clearing router settings...
2022-12-20 17:47:03.175046079 +0000 UTC: Starting link monitor...
2022-12-20 17:47:03.175475542 +0000 UTC: Engine created.
```

/cc @jwhited @raggi @DentonGentry

**Comment by raggi** [comment]
Oh no. I guess this is probably down in x/ somewhere, I have a hard time imagining this is coming from the kernel?

**Comment by jwhited** [comment]
what's the DSM & kernel ver?

**Comment by bradfitz** [comment]
DSM6.

```
admin@ds212j:~$ uname -a
Linux ds212j 2.6.32.12 #25556 Thu Jul 1 14:25:48 CST 2021 armv5tel GNU/Linux synology_88f6281_212j
```

**Comment by bradfitz** [comment]
https://cs.opensource.google/go/x/net/+/refs/tags/v0.4.0:internal/socket/sys_linux_arm.go
```
const (
	sysRECVMMSG = 0x16d
	sysSENDMMSG = 0x176
)
```

https://chromium.googlesource.com/chromiumos/docs/+/master/constants/syscalls.md#arm-32_bit_EABI

365 (0x16d) is just recvmmsg and not some newer variant.

Oh, but the manpage says:

```
VERSIONS       
       The recvmmsg() system call was added in Linux 2.6.33.  Support in
       glibc was added in version 2.12.
```

**Comment by bradfitz** [comment]
Testing options now with:

```
bradfitz@tsdev:~/src/tailscale.com$ SYNO_HOST=10.2.0.22 SYNO_ARCH=arm/5 SYNO_DSM=6 make pushspk
....
tailscale.spk                                                                                                                 100%   21MB   9.5MB/s   00:02    
ssh root@10.2.0.22 /usr/syno/bin/synopkg install tailscale.spk
tailscale.spk install successfully
```

**Comment by bradfitz** [comment]
Fixed in 1.35.89+ for now. Not yet backported.
